sdks/unreal/Agones/Source/Agones/Private/AgonesComponent.cpp

```diff
@@ -0,0 +1,496 @@
+#include "AgonesComponent.h"
+
+#include "Engine/World.h"
+#include "HttpModule.h"
+#include "Interfaces/IHttpResponse.h"
+#include "JsonUtilities/Public/JsonObjectConverter.h"
+#include "TimerManager.h"
+
+UAgonesComponent::UAgonesComponent()
+{
+	PrimaryComponentTick.bCanEverTick = false;
+}
+
+void UAgonesComponent::BeginPlay()
+{
+	Super::BeginPlay();
+	HealthPing(HealthRateSeconds);
+
+	if (bDisableAutoConnect)
+	{
+		return;
+	}
+	Connect();
+}
+
+void UAgonesComponent::EndPlay(const EEndPlayReason::Type EndPlayReason)
+{
+	Super::EndPlay(EndPlayReason);
+	GetWorld()->GetTimerManager().ClearTimer(HealthTimerHandler);
+	GetWorld()->GetTimerManager().ClearTimer(ConnectDelTimerHandle);
+}
+
+TSharedRef<IHttpRequest> UAgonesComponent::BuildAgonesRequest(const FString Path, const FHttpVerb Verb, const FString Content)
+{
+	FHttpModule* Http = &FHttpModule::Get();
+	TSharedRef<IHttpRequest> Request = Http->CreateRequest();

```

Comment:
Any particular reason automatic request retries have been removed? That's probably fine if you have autoConnect turned on but in the case you have it disabled and the game itself calls `Ready` at some point it would be helpful if that request will retry until e.g. the agones sidecar is ready

Reply:
We do have helper functions that do this on startup that can be used to poll the endpoint till the sidecar is healthy, it will then call ready and return the game-server as part of a delegate that you can do with what you need. 

Main reasons for not using the auto request retries was I've not seen it heavily used in other plugins and now that we have the fail call back its much easier to allow the game play programmers to deal with the inability to make the call themselves via delegates, there may be edge cases where they would prefer a failed call to do something else rather than keep retrying.

Reply:
Ah right instead of calling `Ready` you'd then use `Connect` to poll until ready, I think that works. Might be worth adding a small section to the docs calling this out as part of migrating from the old plugin to this one

Reply:
This is more inline with what I have seen in the other Agones plugins from what I remember. Will add a section to the docs.

Reply:
Ahhh, it's already in the docs :D saying that there are additional utility methods that have been added to the plugin that can be used for both health and connect 🥇